apps/src/lab2/views/dialogs/GenericConfirmationDialog.tsx

```diff
@@ -5,7 +5,7 @@ import GenericDialog, {GenericDialogProps} from './GenericDialog';
 export type GenericConfirmationDialogProps = Required<
   Pick<GenericDialogProps, 'title' | 'message'>
 > & {
-  handleConfirm: () => void;
+  handleConfirm?: () => void;
```

Comment:
Since we can now handle post-processing via the promise, the callbacks are now all optional.

There's a future world where we may remove the callbacks entirely and only interact with it via the async promises, but that's something that I didn't want to enforce w/o discussion.